### PR TITLE
[Helm]: Fix metamonitoring secret configuration

### PR DESCRIPTION
**Pull request by Logiraptor**
#### What this PR does

This fixes a regression caused by https://github.com/grafana/mimir/pull/2936.

Without this change, the agent operator is unable to find the usernames, because they aren't written to the secret correctly.

This bug was released in 3.2.0, and since this breaks metamonitoring completely, I suggest we cut a 3.2.1 release with this fix included.

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,8 @@ Entries should include a reference to the Pull Request that introduced the chang
 
 ## main / unreleased
 
+* [BUGFIX] Fix an issue causing metamonitoring secrets to be created incorrectly #3170
```
Comment (osg-grafana) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:28]:
```suggestion
* [BUGFIX] Fix an issue that caused metamonitoring secrets to be created incorrectly #3170
```

**Review by osg-grafana — APPROVED** [review]
Approving from docs, with a minor nit

**Review by krajorama — APPROVED** [review]

**Comment by krajorama** [comment]
3.2 is not out yet, so I'd backport this to release-2.4 branch

**Comment by dimitarvdimitrov** [comment]
thanks for fixing this. 

we should probably be rendering these in the golden record so we catch this earlier. I opened https://github.com/grafana/mimir/pull/3185 to add more config to the offline test metamonitoring-values

**Pull request merged**